File: move_all_exclamation_marks_to_the_end_of_the_sentence.java
public class Kata {
   public static String remove(String s){
       StringBuilder sb=new  StringBuilder();
     int cnt=0;
     for(Character c:s.toCharArray()){
       switch(c){
           case '!'-> ++cnt;
           default -> sb.append(c);
       }
     }
     while(cnt--!=0) sb.append("!");
     return sb.toString();
   }
}

////////////////
interface Kata {
  static String remove(String s) {
    return s.replace("!", "") + s.replaceAll("[^!]", "");
  }
}

///////////
public class Kata {
   public static String remove(String s){
       return s.replaceAll("!", "") + s.replaceAll("[^!]", "");
   }
}

File: c/move_all_exclamation_marks_to_the_end_of_the_sentence.c
char *move_marks (const char *in, char *out)
{
	char* p=out;
  int cnt=0;
  for(;*in;++in){
    if(*in=='!'){
      ++cnt;
    }else{
      *p++=*in;
    }
  }
  while(cnt--){
    *p++='!';
  }
  *p='\0';
	return out; 
}

////////////////
